gui/test/integration/cypress/support/page_files/NodePage.ts

```diff
@@ -0,0 +1,24 @@
+import {Frame} from "./Frame";
+
+export class NodePage extends Frame {
```

Comment:
The reason for the Page superclass is that the way you have it violates the Liskov substitution principle (the "L" in SOLID). A NodePage is not a Frame (! is-a Frame). A NodePage has-a Frame.